lib/cheerio.js

```diff
@@ -77,8 +77,8 @@ var Cheerio = (module.exports = function (selector, context, root, options) {
   }
 
   // If we don't have a context, maybe we have a root, from loading
-  if (!context) {
-    context = this._root;
+  if (!context && this._root && this._root.length) {
+    context = Cheerio.merge(this, this._root);
```

Comment:
Not sure I fully understand this change.

Reply:
Main coal was prevent object `_root` be returned. if you save it directly to context it will returned later and `this` and `_root` will be actually be same object. This way only data is transferred.

Reply:
if context was not provided, but root was provided - `this` becomes context + we add data from root. So even when `_root` are still Cheerio object like before they are not same object.

Reply:
`this` should always be empty (see the `return` in L76), so I don't think this is necessary.

Reply:
I am not sure I understand: `this` on L76 is no longer empty. but starting from L80 we start  building `context` variable. In the end  `context` has to be cheerio object or we cant execute `.find` and in the process empty `this` is very much discarded (L101).

Reply:
On L80, `this` shouldn't have any values, as we immediately return after the only time where elements are added to it (on L76). That means this is equivalent:

```suggestion
  if (!context) {
    context = this._root;
```

Reply:
hmm, actually it is not equivalent, `this` actually becomes clone of `this._root`, but so far no one complains so I relent.